src/rt/sched/backpressure.h

```diff
@@ -0,0 +1,130 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+#pragma once
+
+#include "ds/bits.h"
+
+namespace verona::rt::backpressure
+{
+  /**
+   * Backpressure bits are contain the following fields:
+   * |31          24|23              8|7      0|
+   * |   pressure   |   load history  |  load  |
+   *
+   * The 8-bit pressure field is 0 when muted and in the range [1, 255]
+   * otherwise.
+   *
+   * The 16-bit load history field is a ring buffer with a capacity for 4
+   * 4-bit entries.
+   *
+   * The 8-bit load field tracks an approximation of a cown's message queue
+   * depth as a count of messages processed between token messages falling out
+   * of the queue. When the token message falls out, the `reset()` function
+   * will push the upper nibble of the load into the load history ring buffer.
+   */
+  using backpressure_t = uint32_t;
+
+  static constexpr uint8_t pressure_shift = 24;
+  static constexpr uint32_t pressure_mask = (uint32_t)0xff << pressure_shift;
+
+  static constexpr uint32_t init = (uint32_t)1 << pressure_shift;
+  static constexpr size_t overload_threshold = 800;
+  static constexpr size_t unmute_threshold = 100;
+
+  /**
+   * Return the current pressure value.
+   */
+  static inline uint8_t pressure(uint32_t bp)
+  {
+    return (bp & pressure_mask) >> pressure_shift;
+  }
+
+  /**
+   * Return true if this cown is currently muted and false otherwise.
+   */
+  static inline uint32_t muted(uint32_t bp)
+  {
+    return pressure(bp) == 0;
+  }
+
+  /**
+   * Calculate the total load accumulated for this cown. This value will be in
+   * the range [0, 1215].
+   */
+  static inline uint32_t load(uint32_t bp)
```

Comment:
nit: `load` to me indicates an action "to load a value". Perhaps `get_load` would be clearer to other compiler engineers as well.